pkgs/development/python-modules/backports_shutil_get_terminal_size/default.nix

```diff
@@ -1,17 +1,25 @@
 { lib
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 , pytest
 , pythonOlder
 }:
 
 if !(pythonOlder "3.3") then null else buildPythonPackage rec {
   pname = "backports.shutil_get_terminal_size";
-  version = "1.0.0";
+  version = "2018-06-30";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "713e7a8228ae80341c70586d1cc0a8caa5207346927e23d09dcbcaf18eadec80";
+  # there have been numerous fixes commited since the initial release.
+  # Most notably fixing a problem where the backport would always return
+  # terminal size 0. See https://trac.sagemath.org/ticket/25320#comment:5.
+  # Unfortunately the maintainer seems inactive and has not responded to
+  # a request for a new release since 2016:
+  # https://github.com/chrippa/backports.shutil_get_terminal_size/issues/7
+  src = fetchFromGitHub {
+    owner = "chrippa";
+    repo = "backports.shutil_get_terminal_size";
+    rev = "159e269450dbf37c3a837f6ea7e628d59acbb96a";
```

Comment:
This commit is from 2016-02-21, so please use this as the version.

Reply:
I thought it was supposed to be the date of inclusion of nixpkgs. Doesn't matter either way, I'll change it.

Reply:
I'm not sure actually, but the date of the commit makes much more sense to me.

Reply:
Please don't forget to also update the commit message.